hudi-client/hudi-client-common/src/main/java/org/apache/hudi/metadata/HoodieBackedTableMetadataWriter.java

```diff
@@ -392,6 +398,12 @@ public void initTableMetadata() {
     }
 
     if (!exists) {
+      if (metadataWriteConfig.isMetadataAsyncIndex()) {
+        // with async metadata indexing enabled, there can be inflight writers
+        MetadataRecordsGenerationParams indexParams = getRecordsGenerationParams();
+        scheduleIndex(indexParams.getEnabledPartitionTypes());
```

Comment:
I don't think we need this code block at all. scheduleIndexer will explicitly call schedule.

Reply:
removed.